src/scripts/main.js

```diff
@@ -1,3 +1,113 @@
 'use strict';
 
-// write code here
+const buttons = Array.from(document.querySelectorAll('button'));
+const tableBody = document.querySelector('.field tbody');
+const trElements = document.querySelector('tr');
+
+buttons.forEach((button) => {
+  button.onclick = function (events) {
+    const target = events.target.className;
+
+    switch (target) {
+      case 'append-row button':
+        appendRow();
+        break;
+
+      case 'remove-row button':
+        removeRow();
+        break;
+
+      case 'append-column button':
+        appendColumn();
+        break;
+
+      case 'remove-column button':
+        removeColumn();
+        break;
+    }
+  };
+});
+
+function appendRow() {
+  const rowCount = tableBody.children.length;
+  const removeButton = document.querySelector('.remove-row');
+
+  if (removeButton.hasAttribute('disabled')) {
+    removeButton.removeAttribute('disabled');
+  }
+
+  if (rowCount < 10) {
+    const tr = document.createElement('tr');
+
+    for (let i = 0; i < trElements.children.length; i++) {
+      const td = document.createElement('td');
+
+      tr.appendChild(td);
+    }
+    tableBody.appendChild(tr);
+  }
+
+  if (rowCount === 9) {
+    document.querySelector('.append-row').setAttribute('disabled', 'true');
+  }
+}
+
+function removeRow() {
+  const rowCount = tableBody.children.length;
+  const appendButton = document.querySelector('.append-row');
+
+  if (rowCount > 2) {
+    const tr = tableBody.lastChild;
+
+    tableBody.removeChild(tr);
+  }
+
+  if (rowCount === 2) {
+    document.querySelector('.remove-row').setAttribute('disabled', 'true');
+  }
+
+  if (appendButton.hasAttribute('disabled')) {
+    appendButton.removeAttribute('disabled');
+  }
+}
+
+function appendColumn() {
+  const tr = document.querySelectorAll('tr');
+  const trLength = document.querySelector('tr').children.length;
+
+  if (trLength > 1) {
+    document.querySelector('.remove-column').removeAttribute('disabled');
+  }
+
+  if (trLength === 9) {
```

Comment:
The condition `trLength === 9` for disabling the append column button should be checked after appending a column, not before. Otherwise, it will not disable the button at the correct time.